testsrc/org/mozilla/javascript/tests/JavaIterableTest.java

```diff
@@ -146,19 +139,23 @@ public void testSet() {
     @Test
     public void testNoIterable() {
         Object o = new Object();
-        String script ="for (var e of value) ;";
+        String script = "for (var e of value) ;";
 
-        assertThrows(EcmaError.class, () -> {
-                runScript(script, o);
-            });
+        assertThrows(
+                EcmaError.class,
+                () -> {
+                    runScript(script, o);
+                });
     }
 
     private Object runScript(String scriptSourceText, Object value) {
-        return ContextFactory.getGlobal().call(context -> {
-            context.setLanguageVersion(Context.VERSION_ES6);
-            Scriptable scope = context.initStandardObjects(global);
-            scope.put("value", scope, Context.javaToJS(value, scope));
-            return context.evaluateString(scope, scriptSourceText, "", 1, null);
-        });
+        return ContextFactory.getGlobal()
+                .call(
+                        context -> {
+                            context.setLanguageVersion(Context.VERSION_ES6);
+                            Scriptable scope = context.newObject(global);
```

Comment:
was `Scriptable scope = context.initStandardObjects(global);`
this is the only code change in this class, which is wrong, as it does not create a new scope, but initializes the standardObjects a second time.
I fixed this also in other tests

Reply:
I haven't tested it, but can that line be removed completely? I don't think it's necessary to create a new object as global already refers to an initialized scope. On the other hand, should a new global scope be initialized on each test instead of reusing the same one? I do see the problem you were trying to fix with the single global instance getting reinitialized on each test.

Reply:
> I haven't tested it, but can that line be removed completely?

for this test case, it would work. Or in other words, it was just like that:
`context.initStandardObjects` returns `global` so the code looks like this:
```java
context.initStandardObjects(global);
Scriptable scope = global;
```

> On the other hand, should a new global scope be initialized on each test instead of reusing the same one?

it depends on the test. If the test itself modifies the scope, it would be better to initialize a new one. In this test, the next line puts `value` into the scope (`scope.put("value", scope, Context.javaToJS(value, scope));`) - as this is done on each test, I would agree, that it is not neccessary in this test, but the origin problem was, that `initStandardObjects` mess up some objects if it is called multiple times